mantle/sdk/enter.go

```diff
@@ -1,545 +0,0 @@
-
-	// User Credential fields
-	ClientSecret string `json:"client_secret"`
-	RefreshToken string `json:"refresh_token"`
-
-	// Service Account fields
-	ClientEmail  string `json:"client_email"`
-	PrivateKeyID string `json:"private_key_id"`
-	PrivateKey   string `json:"private_key"`
-}
-
-// MountAPI mounts standard Linux API filesystems.
-// When possible the filesystems are mounted read-only.
-func (e *enter) MountAPI() error {
-	var apis = []struct {
-		Path string
-		Type string
-		Opts string
-	}{
-		{"/proc", "proc", "ro,nosuid,nodev,noexec"},
-		{"/sys", "sysfs", "ro,nosuid,nodev,noexec"},
-		{"/run", "tmpfs", "nosuid,nodev,mode=755"},
-	}
-
-	// Make sure the new root directory itself is a mount point.
-	// `unshare` assumes that `mount --make-rprivate /` works.
-	if err := system.RecursiveBind(e.Chroot, e.Chroot); err != nil {
-		return err
-	}
-
-	for _, fs := range apis {
-		target := filepath.Join(e.Chroot, fs.Path)
-		if err := system.Mount("", target, fs.Type, fs.Opts); err != nil {
-			return err
-		}
-	}
-
-	// Since loop devices are dynamic we need the host's managed /dev
-	if err := system.ReadOnlyBind("/dev", filepath.Join(e.Chroot, "dev")); err != nil {
-		return err
-	}
-	// /dev/pts must be read-write because emerge chowns tty devices.
-	if err := system.Bind("/dev/pts", filepath.Join(e.Chroot, "dev/pts")); err != nil {
-		return err
-	}
-
-	// Unfortunately using the host's /dev complicates /dev/shm which may
-	// be a directory or a symlink into /run depending on the distro. :(
-	// XXX: catalyst does not work on systems with a /dev/shm symlink!
-	if system.IsSymlink("/dev/shm") {
-		shmPath, err := filepath.EvalSymlinks("/dev/shm")
-		if err != nil {
-			return err
-		}
-		// Only accept known values to avoid surprises.
-		if shmPath != "/run/shm" {
-			return fmt.Errorf("Unexpected shm path: %s", shmPath)
-		}
-		newPath := filepath.Join(e.Chroot, shmPath)
-		if err := os.Mkdir(newPath, 01777); err != nil {
-			return err
-		}
-		if err := os.Chmod(newPath, 01777); err != nil {
-			return err
-		}
-	} else {
-		shmPath := filepath.Join(e.Chroot, "dev/shm")
-		if err := system.Mount("", shmPath, "tmpfs", "nosuid,nodev"); err != nil {
-			return err
-		}
-	}
-
-	return nil
-}
-
-// MountAgent bind mounts a SSH or GnuPG agent socket into the chroot
-func (e *enter) MountSSHAgent() error {
-	origPath := os.Getenv("SSH_AUTH_SOCK")
-	if origPath == "" {
-		return nil
-	}
-
-	origDir, origFile := filepath.Split(origPath)
-	if _, err := os.Stat(origDir); err != nil {
-		// Just skip if the agent has gone missing.
-		return nil
-	}
-
-	newDir, err := ioutil.TempDir(e.UserRunDir, "agent-")
-	if err != nil {
-		return err
-	}
-
-	if err := system.Bind(origDir, newDir); err != nil {
-		return err
-	}
-
-	newPath := filepath.Join(newDir, origFile)
-	chrootPath := strings.TrimPrefix(newPath, e.Chroot)
-	return os.Setenv("SSH_AUTH_SOCK", chrootPath)
-}
-
-// MountGnupg bind mounts $GNUPGHOME or ~/.gnupg and the agent socket
```

Comment:
Note that since this code was created in 2015 we've basically made this concept of "run some code unprivileged in a container sharing my home directory" a [first class concept](https://github.com/containers/toolbox).